File: Week1/remove-duplicates-from-sorted-list-ii/tangxinye_solution2.cpp
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if (!head || !head->next) return head;
        ListNode *now = head->next;
        if (head->val == now->val) {
            while (now && head->val == now->val) {
                ListNode *tmp = now;
                now = now->next;
                delete tmp;
            }
            delete head;
            return deleteDuplicates(now);
        } else {
            head->next = deleteDuplicates(head->next);
            return head;
        }
    }
};

File: Week1/remove-duplicates-from-sorted-list-ii/tangxinye_solution2.java
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null||head.next==null)
        	return head;
        
        ListNode now= head.next;
        if(head.val==now.val){
        	while(now!=null&&head.val==now.val){
        		now=now.next;
        	}
        	return deleteDuplicates(now);
        }else {
			head.next=deleteDuplicates(head.next);
		}
        return head;
    }
}
